python/pyspark/sql/profiler.py

```diff
@@ -224,6 +224,54 @@ def dump(id: int) -> None:
             for id in sorted(code_map.keys()):
                 dump(id)
 
+    def clear_perf_profiles(self, id: Optional[int] = None) -> None:
+        """
+        Clear the perf profile results.
+
+        .. versionadded:: 4.0.0
```

Comment:
Is this a user-facing API? If not, we don't need this version directive

Reply:
It is a user-facing API, along with `profile.show` and `profile.dump`. We will also add it to API doc.

Reply:
Actually this is not. The `clear` in `Profile` should be a user-facing API.